projectq/backends/_azure/_azure_quantum.py

```diff
@@ -0,0 +1,403 @@
+                    logical_id = tag.logical_qubit_id
+                    break
+
+            if logical_id is None:
+                raise RuntimeError('No LogicalQubitIDTag found in command!')
+
+            self._measured_ids.append(logical_id)
+            return
+
+        if self._provider_id == IONQ_PROVIDER_ID:
+            if not self._circuit:
+                self._circuit = []
+
+            json_cmd = to_json(cmd)
+            if json_cmd:
+                self._circuit.append(json_cmd)
+
+        elif self._provider_id == QUANTINUUM_PROVIDER_ID:
+            if not self._circuit:
+                self._circuit = ''
+
+            qasm_cmd = to_qasm(cmd)
+            if qasm_cmd:
+                self._circuit += '\n{0}'.format(qasm_cmd)
+
+    def is_available(self, cmd):
+        """
+        Test if this backend is available to process the provided command.
+
+        Args:
+            cmd (Command): A command to process.
+
+        Returns:
+            bool: If this backend can process the command.
+        """
+        if self._provider_id == IONQ_PROVIDER_ID:
+            return is_available_ionq(cmd)
+        elif self._provider_id == QUANTINUUM_PROVIDER_ID:
+            return is_available_quantinuum(cmd)
+
+        return False
+
+    @property
+    def _target_factory(self):
+        target_factory = TargetFactory(
+            base_cls=Target,  # noqa
+            workspace=self._workspace,
+            default_targets=AzureQuantumBackend.DEFAULT_TARGETS
+        )
+
+        return target_factory
+
+    @property
+    def _target(self):
+        target = self._target_factory.get_targets(
+            name=self._target_name,
+            provider_id=self._provider_id
+        )
+
+        if type(target) is list and len(target) == 0:  # pragma: no cover
+            raise AzureQuantumTargetNotFoundError(
+                'Target {} is not available on workspace {}.'.format(
+                    self._target_name, self._workspace.name)
+            )
+
+        return target
+
+    @property
+    def current_availability(self):
+        """Current availability for given target."""
+        return self._target.current_availability
+
+    @property
+    def average_queue_time(self):
+        """Average queue time for given target."""
+        return self._target.average_queue_time
+
+    def get_probability(self, state, qureg):
+        """Shortcut to get a specific state's probability.
+
+        Args:
+            state (str): A state in bit-string format.
+            qureg (Qureg): A ProjectQ Qureg object.
+
+        Returns:
+            float: The probability for the provided state.
+        """
+        if len(state) != len(qureg):
+            raise ValueError('Desired state and register must be the same length!')
+
+        probs = self.get_probabilities(qureg)
+
+        try:
+            return probs[state]
+        except KeyError:
+            return 0.0
```

Comment:
```suggestion
        return probs.get(state, 0.0)
```

Reply:
I can't believe I missed this :). This is updated.